### PR TITLE
Add endpoint for retrieving GlossaryDocs

### PR DESCRIPTION
**Pull request by GoddessEyes**
* add `get_glossary_doc` query
* refactor schema (remove `GlossaryDocumentListResponse`)
* change glossary listing endpoint
* add `BaseQueryException`

### CODE REVIEW

File: backend/app/routers/glossary.py [hunk 1]
```diff
@@ -18,14 +29,39 @@
...
 )
 def list_glossary_docs(db: Session = Depends(get_db)):
-    glossaries = list_glossary_docs_controller(db)
-    return GlossaryDocumentListResponse(glossaries=glossaries)
+    return list_glossary_docs_controller(db)
+
+
+@router.get(
+    "/docs/{glossary_doc_id}",
+    description="Get single glossary documents",
```
Comment (ArchiDevil) [inline backend/app/routers/glossary.py:43]:
```suggestion
    description="Get a single glossary document",
```

File: backend/app/glossary/controllers.py [hunk 2]
```diff
@@ -19,4 +20,15 @@ def create_glossary_doc_from_file_controller(
...
+    glossaries = GlossaryDocsQuery(db).list_glossary_docs()
+    return [
+        GlossaryDocumentResponse.model_validate(glossary) for glossary in glossaries
+    ]
+
+
+def retrieve_glossary_doc_controller(glossary_doc_id: int, db: Session):
+    try:
+        doc = GlossaryDocsQuery(db).get_glossary_doc(glossary_doc_id)
+        return GlossaryDocumentResponse.model_validate(doc)
```
Comment (ArchiDevil) [inline backend/app/glossary/controllers.py:32]: What is the purpose of model validation here after requesting it from DB?

**Review by ArchiDevil — CHANGES_REQUESTED** [review]

File: backend/app/glossary/controllers.py (hunk 2 above)
Comment (GoddessEyes) [reply backend/app/glossary/controllers.py:32]: Validate a pydantic model instance
https://docs.pydantic.dev/latest/api/base_model/#pydantic.BaseModel.model_validate

Cast from DbModel instance to Pydantic instance

File: backend/app/routers/glossary.py (hunk 1 above)
Comment (GoddessEyes) [reply backend/app/routers/glossary.py]: Fixed

**Review by ArchiDevil — APPROVED** [review]

**Pull request merged**